src/js/components/Button/Button.js

```diff
@@ -343,9 +481,36 @@ const Button = forwardRef(
     }
     if (tip) {
       if (typeof tip === 'string') {
-        return <Tip content={tip}>{styledButtonResult}</Tip>;
+        styledButtonResult = <Tip content={tip}>{styledButtonResult}</Tip>;
       }
-      return <Tip {...tip}>{styledButtonResult}</Tip>;
+      styledButtonResult = <Tip {...tip}>{styledButtonResult}</Tip>;
+    }
+
+    // if the caller doesn't specify that they want
+    // the badge relative to the button content, set the badge
+    // relative to the outer edge of the button
+    if (badgeProp && badgeProp.target !== 'contents') {
+      const badge = getBadge(
```

Comment:
There's a fair amount of code duplication here. Wondering if there's a way to restructure things to avoid that?

Reply:
Yes, let me think about how to minimize duplication.